src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -0,0 +1,708 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class DelegateTypeTests : CSharpTestBase
+    {
+        [Fact]
+        public void LanguageVersion()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        System.Delegate d;
+        d = Main;
+        d = () => { };
+        d = delegate () { };
+    }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (6,13): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         d = Main;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "Main").WithArguments("inferred delegate type").WithLocation(6, 13),
+                // (7,13): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         d = () => { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "() => { }").WithArguments("inferred delegate type").WithLocation(7, 13),
+                // (8,13): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         d = delegate () { };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "delegate () { }").WithArguments("inferred delegate type").WithLocation(8, 13));
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LambdaConversions_01()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        object o = () => { };
+        System.ICloneable c = () => { };
+        System.Delegate d = () => { };
```

Comment:
Should we also test MulticastDelegate per LDM discussion yesterday?

Reply:
PROTOTYPE for `object` and `ICloneable` (those should work too, right?)

---
In reply to: [612747777](https://github.com/dotnet/roslyn/pull/52448#discussion_r612747777) [](ancestors = 612747777)

Reply:
Added failing test for `System.MulticastDelegate` and added items to the test plan for each (rather than adding PROTOTYPE comments).